src/Compilers/VisualBasic/Portable/Compilation/VisualBasicDiagnosticFilter.vb

```diff
@@ -176,6 +176,11 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
             End If
 
+            Dim globalReport As ReportDiagnostic
+            If (Not isSpecified) AndAlso syntaxTreeOptions IsNot Nothing AndAlso syntaxTreeOptions.TryGetGlobalDiagnosticValue(id, globalReport) Then
```

Comment:
Same suggestion here. Rule ID specific global config options should override Rule ID agnostic options.